core/src/main/java/io/grpc/internal/JsonParser.java

```diff
@@ -56,13 +56,32 @@ public static Object parse(String raw) throws IOException {
     }
   }
 
-  private static Object parseRecursive(JsonReader jr) throws IOException {
+  /**
+   * Parses a json string, returning either a {@code Map<String, ?>}, {@code List<?>},
+   * {@code String}, {@code Double}, {@code Boolean}, or {@code null}. Fails if duplicate names
+   * found.
+   */
+  public static Object parseNoDuplicates(String raw) throws IOException {
```

Comment:
I thought we were going to change the code to do this all the time. C was doing it already, so it is already reasonably safe, and if you have duplicates you really don't know what result you'll get.

If you are concern about this causing problems, then we could have a flag set from an environment variable. We default it to "fail on duplicate" but someone can change it if it is a problem. After a few releases, we drop the old way.

Reply:
Yeah, changing the default behavior would be ideal. Unfortunately I ran into an issue with
https://github.com/grpc/grpc-java/blob/master/gcp-observability/src/test/java/io/grpc/gcp/observability/ObservabilityConfigImplTest.java#L109
Does Observability conf allow duplicates?